File: Python/Day17_14_05_2025/HierarchicalInheritance.py
""" Q4. Hierarchical Inheritance:
Create a base class Shape with a method area() that prints "Area not defined." 
Create two derived classes, Circle and Rectangle.
The circle should calculate and print the area using the radius.
The rectangle should calculate and print the area using length and width."""
import math
class Shape:
    def area(self):
        print("Area not defined.")
class Circle(Shape):
    def __init__(self, radius):
        self.radius = radius
    def area(self):
        area = math.pi * self.radius ** 2
        print(f"Circle Area: {area:.2f}")
class Rectangle(Shape):
    def __init__(self, length, width):
        self.length = length
        self.width = width
    def area(self):
        area = self.length * self.width
        print(f"Rectangle Area: {area}")
circle = Circle(radius=5)
rectangle = Rectangle(length=4, width=6)
circle.area()      
rectangle.area()   

File: Java/Day11_14_05_2025/src/HierarchicalInheritance.java
/* Q3.⁠ ⁠Hierarchical Inheritance
Create a class Shape with a method area() that prints "Calculating area." Derive two classes:
Circle with a method radiusArea(double radius) that calculates the area of a circle.
Rectangle with a method rectArea(double length, double width) that calculates the area of a rectangle.
Task: Create objects of both Circle and Rectangle classes and calculate areas for given inputs. */
class Shape {
    void area() {
        System.out.println("Calculating area.");
    }
}
class Circle extends Shape {
    void radiusArea(double radius) {
        double area = Math.PI * radius * radius;
        System.out.println("Area of Circle: " + area);
    }
}
class Rectangle extends Shape {
    void rectArea(double length, double width) {
        double area = length * width;
        System.out.println("Area of Rectangle: " + area);
    }
}
public class HierarchicalInheritance {
    public static void main(String[] args) {
        Circle circle = new Circle();
        Rectangle rectangle = new Rectangle();
        circle.area();
        circle.radiusArea(5);

        rectangle.area();
        rectangle.rectArea(4, 7);
    }
}